napari/utils/translations.py

```
@@ -0,0 +1,456 @@
+        stdout, stderr = p.communicate()
+        result = json.loads(stdout.decode('utf-8'))
+    except Exception:
+        result["message"] = (
+            traceback.format_exc() + "\n" + repr(stderr.decode('utf-8'))
+        )
+
+    return result["data"], result["message"]
+
+
+def get_installed_packages_locales() -> dict:
+    """
+    Get all napari extensions installed that contain locale data.
+
+    Returns
+    -------
+    dict
+        Ordered list of available language packs.
+        >>> {"package-name": locale_data, ...}
+
+    Examples
+    --------
+    - `entry_points={"napari.locale": "package-name = package_module"}`
+    - `entry_points={"napari.locale": "napari-git = napari_git"}`
+    """
+    cmd = [sys.executable, __file__, "_get_installed_language_pack_locales"]
+    found_package_locales, message = _run_process_and_parse(cmd)
+    return found_package_locales, message
+
+
+# --- Translators
+# ----------------------------------------------------------------------------
+class TranslationBundle:
+    """
+    Translation bundle providing gettext translation functionality.
+    """
+
+    def __init__(self, domain: str, locale: str):
+        self._domain = domain
+        self._locale = locale
+
+        self._update_locale(locale)
+
+    def _update_locale(self, locale: str):
+        """
+        Update the locale environment variables.
+
+        Parameters
+        ----------
+        locale : str
+            The language name to use.
+        """
+        self._locale = locale
+        localedir = None
+        if locale.split("_")[0] != DEFAULT_LOCALE:
+            data, _ = get_installed_packages_locales()
+            language_pack_module = data.get(locale)
+            try:
+                if language_pack_module is not None:
+                    mod = importlib.import_module(language_pack_module)
+                    localedir = Path(mod.__file__).parent / LOCALE_DIR
+            except ImportError:
+                pass
```

does this just fail silently if someone requests an unavailable locale?  Should we show a warning?

---

We could, what is the normal way of rasining an error on napari?

Any convention?